File: SjoslagetBackend/Models/Payment.cs
using System;

namespace Accidis.Sjoslaget.WebService.Models
{
	public sealed class Payment
	{
		public Guid Id { get; set; }
		public Guid BookingId { get; set; }
		public decimal Amount { get; set; }
		public DateTime Created { get; set; }
	}
}

File: FrontendShared/lib/model/payment.dart
import 'dart:convert';

import 'package:decimal/decimal.dart';
import 'package:frontend_shared/util.dart';

import 'json_field.dart';

class Payment {
	final Decimal amount;
	final DateTime created;

	Payment(this.amount, this.created);

	factory Payment.fromMap(Map<String, dynamic> json) =>
		Payment(
			ValueConverter.doubleToDecimal(json[AMOUNT]),
			DateTime.parse(json[CREATED])
		);
}
